linmod/data.py

```diff
@@ -201,6 +201,55 @@
 """
 
 
+class CountsFrame(pl.DataFrame):
+    """
+    A `polars.DataFrame` which enforces a format for observed counts of lineages.
+
+    See `REQUIRED_COLUMNS` for the expected columns.
+    """
+
+    REQUIRED_COLUMNS = {
+        "date",
+        "fd_offset",
+        "division",
+        "lineage",
+        "count",
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.validate()
+
+    @classmethod
+    def read_parquet(cls, *args, **kwargs):
+        return cls(pl.read_parquet(*args, **kwargs))
+
+    def validate(self, *args, **kwargs):
+        # In case polars ever adds a validate method
+        if hasattr(super(), "validate"):
+            super().validate(*args, **kwargs)
+
+        assert self.REQUIRED_COLUMNS.issubset(
+            self.columns
+        ), f"Missing at least one required column ({', '.join(self.REQUIRED_COLUMNS)})"
```

Comment:
```suggestion
        ), f"Missing required columns: ({', '.join(req for req in self.REQUIRED_COLUMNS if not req in self.columns)})"
```

We could be more descriptive. Not sure it really helps much though

Reply:
Also now a big fan of https://docs.pola.rs/api/python/stable/reference/schema/index.html, e.g. so you could say `my_data_frame.schema.to_python()`

`df.columns` is easier than `df.schema.names()`, but it's something to keep in mind, when doing these kinds of checks